apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx

```diff
@@ -282,20 +338,30 @@ function EnrollForm(props) {
       taught_ap_before: formState.taught_ap_before,
       planning_to_teach_ap: formState.planning_to_teach_ap,
     };
-    $.ajax({
-      method: 'POST',
-      url: `/api/v1/pd/workshops/${props.workshop_id}/enrollments`,
-      contentType: 'application/json',
-      data: JSON.stringify(params),
-      complete: result => {
-        setIsSubmitting(false);
-        result?.responseJSON?.workshop_enrollment_status === 'error' &&
-          setSubmissionErrorMessage(
-            result?.responseJSON?.error_message || 'unknown error'
-          );
+
+    try {
+      const response = await fetch(
+        `/api/v1/pd/workshops/${props.workshop_id}/enrollments`,
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            'X-CSRF-Token': await getAuthenticityToken(),
+          },
+          body: JSON.stringify(params),
+        }
+      );
+      setIsSubmitting(false);
+      if (response.ok) {
+        const result = await response.json();
+        if (result.workshop_enrollment_status === 'error') {
+          setSubmissionErrorMessage(result.error_message || 'unknown error');
+        }
         props.onSubmissionComplete(result);
-      },
-    });
+      }
+    } catch (error) {
+      setSubmissionErrorMessage('unknown error');
```

Comment:
Do we want to log what the error is in some way instead of just defaulting all errors to "unknown error"s?

Reply:
good question! I don't really want to log to the console, but is there some client side error reporting we can use? would that be statsig? is honeybadger only for the back end?

prior to this refactor there was not any error handling on the request itself, only errors in the response object. at least from what I could tell 🤔

Reply:
Oh gotcha, if you're keeping the same behavior then that's fine! It should still go to Honeybadger if there's an error!